### PR TITLE
Add __str__ methods for Error classes

### PR DESCRIPTION
**Pull request by djc**
This helps diagnosing errors.

### CODE REVIEW

**Comment by alex** [comment]
Can we add a small test for these?

**Comment by djc** [comment]
Like that? (Rebased a bit to fix a bug in the initial commit.)

**Comment by djc** [comment]
@alex would you mind pronouncing on this? I've been doing work on one of my rply-based projects again and this PR still made it easier to pinpoint lexer errors in particular.

**Comment by alex** [comment]
These should probably be `__repr__`, instead of `__str__`. But otherwise LGTM.

File: tests/test_lexer.py [hunk 1]
```diff
@@ -137,3 +137,16 @@ def test_ignore_recursion(self):
...
+        lg.add("NUMBER", r"\d+")
+        lg.add("PLUS", r"\+")
+
+        l = lg.build()
+
+        stream = l.lex('fail')
+        with raises(LexingError) as excinfo:
+            stream.next()
+
+        assert 'SourcePosition(' in str(excinfo.value)
```
Comment (alex) [inline tests/test_lexer.py]: `s/str/repr/`

File: tests/test_parser.py [hunk 2]
```diff
@@ -164,6 +164,7 @@ def main(p):
             ]))
 
         assert exc_info.value.getsourcepos().lineno == 10
+        assert 'SourcePosition' in str(exc_info.value)
```
Comment (alex) [inline tests/test_parser.py]: Same here

**Comment by djc** [comment]
Should be all good now.

**Pull request merged**